.github/workflows/master.yml

```diff
@@ -0,0 +1,135 @@
+    steps:
+      - uses: actions/checkout@v2
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(python matrix.py)"
+
+  run-tests:
+    needs: support-matrix
+    name: Test Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ${{ fromJson(needs.support-matrix.outputs.matrix).os }}
+        python-version: ${{ fromJson(needs.support-matrix.outputs.matrix).python-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements.dev.txt') }}
+      - name: Install Dependencies
+        run: |
+          pip install --upgrade --upgrade-strategy eager -r requirements.txt -r requirements.dev.txt -e .
+      - name: Run Tests
+        uses: ./.github/workflows/run-tests
+        with:
+          coverage-artifact-name: ${{ github.workflow }}
+
+
+      # --- Split steps below into separate jobs once caching is implemented
+
+      # --- job: publish-docs
+      - name: Install Doc Dependencies
+        uses: ./.github/workflows/install-dependencies
+        if: (runner.os == 'Linux') && (matrix.python-version == '3.8')
+        with:
+          requirements: requirements.publish.txt
+          conda-channel: conda-forge
+      - name: Publish Docs
+        shell: bash -l {0}
+        if: (runner.os == 'Linux') && (matrix.python-version == '3.8')
+        run: |
+          pip install travis-sphinx
+          rm -rf docs/source/*.txt
+          SPHINX_APIDOC_OPTIONS=members,undoc-members,show-inheritance sphinx-apidoc -eM -s txt -o docs/source/ exatomic *test*
+          travis-sphinx build
+          travis-sphinx deploy
+
+      # --- job: publish-artifacts (maybe separate github / conda)
+      # --- need the following secrets from the project
+      # ------ secrets.GITHUB_TOKEN
+      # ------ secrets.PYPI_PASSWORD
+      # ------ secrets.ANACONDA_USERNAME
+      # ------ secrets.ANACONDA_PASSWORD
+      - name: Install Publish Dependencies
+        uses: ./.github/workflows/install-dependencies
+        with:
+          requirements: requirements.release.txt
+          conda-channel: defaults
+      - name: Publish Artifacts
+        if: runner.os == 'Linux'
+        run: |
+          export PYTHONVER="${{ matrix.python-version }}"
+          export pyver="${PYTHONVER/./}"
+          export exver=`cat exatomic/static/version.txt`
+          git remote set-url origin https://${{ secrets.GITHUB_TOKEN }}@github.com/exa-analytics/exatomic.git
+          git tag ${exver}
+          git push --tags
+          sed -i "s/version = .* /version = \"${exver}\"/" meta.yaml
```

Comment:
Can (should we?) cook in an auto-increment to the version number? I'm think we stay semantic `major.minor.micro` and always bump micro during publish and leave manual bumps for major (backwards incompatible) and minor (new features)?

Reply:
Mm yeah `master.yml` still needs some work. I think with actions there's no need for the GITHUB_TOKEN, but yes a sane approach for version incrementing is in order. I like the idea of an auto-micro version bump and then manual minor and major version bumps. I have vague memories of the use of versioneer somewhere but perhaps we can just follow the outlined approach.